rdbms/src/main/java/io/spine/server/storage/jdbc/record/WriteEntityQuery.java

```diff
@@ -0,0 +1,164 @@
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.storage.jdbc.record;
+
+import com.google.common.base.Functions;
+import com.querydsl.core.dml.StoreClause;
+import io.spine.server.entity.storage.ColumnRecords;
+import io.spine.server.entity.storage.ColumnTypeRegistry;
+import io.spine.server.entity.storage.EntityRecordWithColumns;
+import io.spine.server.storage.jdbc.query.AbstractQuery;
+import io.spine.server.storage.jdbc.query.IdColumn;
+import io.spine.server.storage.jdbc.query.Parameters;
+import io.spine.server.storage.jdbc.query.WriteQuery;
+import io.spine.server.storage.jdbc.type.JdbcColumnType;
+import io.spine.server.storage.jdbc.type.JdbcTypeRegistryFactory;
+
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.collect.Maps.newLinkedHashMap;
+import static io.spine.server.storage.jdbc.query.Serializer.serialize;
+import static io.spine.server.storage.jdbc.record.RecordTable.StandardColumn.entity;
+
+/**
+ * An abstract base for write queries to the {@link RecordTable}.
+ *
+ * <p>The query can be used to work with {@linkplain Builder#addRecords(Map) multiple records}.
+ *
+ * <p>An overhead for multiple records will be the same as for a single record.
+ *
+ * @param <I> the type of IDs
+ * @param <C> the type of {@link StoreClause}
+ * @author Dmytro Grankin
+ */
+abstract class WriteEntityQuery<I, C extends StoreClause<C>> extends AbstractQuery implements WriteQuery {
+
+    private final IdColumn<I> idColumn;
+    private final Map<I, EntityRecordWithColumns> records;
+    private final ColumnTypeRegistry<? extends JdbcColumnType<?, ?>> columnTypeRegistry;
+
+    WriteEntityQuery(Builder<? extends Builder, ? extends WriteEntityQuery, I> builder) {
+        super(builder);
+        this.idColumn = builder.idColumn;
+        this.records = builder.records;
+        this.columnTypeRegistry = builder.columnTypeRegistry;
+    }
+
+    @Override
+    public long execute() {
+        final C clause = createClause();
+        for (I id : records.keySet()) {
+            final EntityRecordWithColumns record = records.get(id);
+            setEntityColumns(clause, record);
+
+            clause.set(pathOf(entity), serialize(record.getRecord()));
+            setIdValue(clause, idColumn, idColumn.normalize(id));
+            addBatch(clause);
+        }
+        return clause.execute();
+    }
+
+    /**
+     * Adds either an {@linkplain com.querydsl.sql.dml.SQLUpdateClause#addBatch() update}
```

Comment:
This documentation and the method design itself are weak in case Querydsl update the number of clauses, that extend `StoreClause`.

Reply:
Done.